dbt/adapters/bigquery/python_submissions.py

```diff
@@ -115,14 +115,17 @@ def _get_job_client(self) -> dataproc_v1.BatchControllerClient:
             client_options=self.client_options, credentials=self.GoogleCredentials
         )
 
+    def _get_batch_id(self) -> str:
+        return self.parsed_model["config"].get("batch_id", self.credential.batch_id)
```

Comment:
extending this idea: if the user does not specify a `batch_id` why not use the model name as the batch id?

Reply:
💡

Reply:
Or just leave the `batch_id` as undeclared (which BigQuery will fill in with a random string generation):

```suggestion
        return self.parsed_model["config"].get("batch_id")
```

Since setting `batch_id` within the profile won't work for projects with more than one Python model, it seems like we should consider reverting https://github.com/dbt-labs/dbt-bigquery/pull/727 and replace it with this model-specific logic instead.

Reply:
@dbeatty10 your suggestion is probably best on this, dataproc doesn't need you to provide a batch_id as part of your request, & if we use the model name instead, there's a chance that the model name wouldn't be a valid batch_id - 

> This value must be 4-63 characters. Valid characters are /[a-z][0-9]-/.

source: https://cloud.google.com/dataproc-serverless/docs/reference/rpc/google.cloud.dataproc.v1#google.cloud.dataproc.v1.Batch